ocsalamanders/src/main/java/com/yfletch/ocsalamanders/OCSalamandersPlugin.java

```diff
@@ -34,14 +40,58 @@ public void init(SalamandersConfig config, SalamandersContext context)
 		refreshOnConfigChange(true);
 	}
 
+	@Override
+	protected void startUp()
+	{
+		super.startUp();
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		super.shutDown();
+		context.clearPrimaryTree();
+	}
+
+	@Subscribe
+	public void OnMenuOptionClicked(MenuOptionClicked menuOptionClicked)
+	{
+		if(context.getPrimaryTree() != null || !menuOptionClicked.getMenuTarget().contains("Young tree")) return;
+		switch(menuOptionClicked.getMenuAction())
+		{
+			case WIDGET_TARGET_ON_GAME_OBJECT:
+			case GAME_OBJECT_FIRST_OPTION:
+			case GAME_OBJECT_SECOND_OPTION:
+			case GAME_OBJECT_THIRD_OPTION:
+			case GAME_OBJECT_FOURTH_OPTION:
+			case GAME_OBJECT_FIFTH_OPTION:
+			{
+				int x = menuOptionClicked.getParam0();
+				int y = menuOptionClicked.getParam1();
+				context.setPrimaryTree(x, y);
+				break;
+			}
+		}
+	}
+
 	@Override
 	public void setup()
 	{
-		requirements().mustBeNear(() -> object("Young tree"));
+		requirements()
+				.mustBeNear(() -> object("Young tree"))
+				.must((ctx) -> ctx.getPrimaryTree() != null, "Click on tree to start");
 
 		action().name("Drop salamander")
-			.when(c -> Inventory.contains("Black salamander"))
-			.then(c -> item("Black salamander").interact("Release"))
+			.when(c -> Inventory.contains(
+					ItemID.BLACK_SALAMANDER,
+					ItemID.RED_SALAMANDER,
+					ItemID.ORANGE_SALAMANDER
+			))
+			.then(c -> item(
+					"Black salamander",
+					"Red salamander",
+					"Orange salamander"
+			).interact("Release"))
```

Comment:
alternative (may require static Util fn import)

```suggestion
			.then(c -> item(containing("salamander")).interact("Release"))
```